### PR TITLE
Connection status handling

### LINKED ISSUE #1099: Blank screen failure caused by inaccessible remote zqd
This issue was originally identified by @orochford. My repro is using GA Brim tagged `v0.18.0` talking to GA `zqd` tagged `v0.22.0`.

First I need to start a remote `zqd` and populate it with a Space. In my case I started Brim in a Linux VM in VirtualBox and imported a pcap. I used `ps` to grab the `zqd` command line that Brim had started, then quit the app and restarted `zqd` standalone by modifying the command line to drop the `localhost` from the `-l` parameter and remove the `-brimfd` from the end. Hence:

```
/usr/lib/brim/resources/app/zdeps/zqd listen -l :9867 -data /home/phil/.config/Brim/data/spaces -config /home/phil/.config/Brim/zqd-config.yaml -zeekrunner /usr/lib/brim/resources/app/zdeps/zeek/zeekrunner
```

I have a port forwarding rule configured in VirtualBox such that `localhost:9999` gets forwarded to port `9867` in the VM.

Now from the local Brim desktop on my Mac, I connect successfully to the remote `zqd` via `localhost:9999`, click the Space name to see its contents, then quit the app.

Now I stop the remote `zqd` and relaunch the app. I get the white screen with DevTools console showing:

image

### PR DESCRIPTION
**Pull request by mason-fish**
fixes #1099 

**Changes**
 * Introduce connection statuses, stored in the clusters state.
 * Persist spaces state.
 * Initialization no longer depends on zqd and instead the data syncs come from the loaded app components

Every time the app loads we reset all statuses to "initial", indicating that we are unsure if the connection will succeed yet, and also indicating that we need to sync with server data. While syncing, if the connection fails we set the status to "disconnected" and display as much in a new tab content page where the user my attempt to retry.

brim

Signed-off-by: Mason Fish <mason@looky.cloud>

### CODE REVIEW

**Review by jameskerr — CHANGES_REQUESTED** [review]
Right on! I like the code. I'm glad we talked and thought about it for a minute. Thanks for deleting all that old login stuff too.

In addition to the suggestions below, here are a few thoughts.

**Yellow Notice**

Now that we have this connection error view, the yellow notice that says "Failed to fetch" is unnecessary. I think it should not appear in this case (when we're testing out a connection). 

Screen Shot 2020-10-15 at 10 41 01 AM

This might be more difficult than it seems. We pop up that "Notice" if any request in the whole app throws. You'll have to see if there is a way to remove the error handler only here, or if we need to remove it from where it is now and place it only where it's needed.

**Wording of notice**


As Al suggested, I think the wording of this error message needs to be updated to:

```
The service at {host}:{port} could not be reached.
```

Screen Shot 2020-10-15 at 10 40 52 AM

File: src/js/components/ConnectionError.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  conn: Cluster
+  retry: Function
+}
+
+const ConnectionError = ({conn, retry}: Props) => {
+  const [isFetching, setIsFetching] = useState(false)
+
+  const onClick = async () => {
+    setIsFetching(true)
+    await retry()
```
[the remark this answers is not in the document]
Comment (mason-fish) [reply src/js/components/ConnectionError.tsx:44]: ah interesting! I had put in a wait just to test that it worked properly but then took it back out. I noticed the missing feedback then and that's why I added the Notice to the retry method, but if we end up removing that then I def agree that some sort of feedback is important

File: src/js/components/ConnectionError.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+const PageWrap = styled.div`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+`
+
+const StyledHeader = styled.h1`
```
[the remark this answers is not in the document]
Comment (mason-fish) [reply src/js/components/ConnectionError.tsx:16]: sure thing

File: src/js/components/ClusterPicker.tsx [hunk 3]
```diff
@@ -46,8 +45,8 @@ export default function ClusterPicker() {
       checked: isCurrent,
       click: () => {
         if (isCurrent) return
-        dispatch(setConnection(c)).catch((e) => {
-          dispatch(Notice.set(e))
+        dispatch(initConnection(c)).catch(() => {
+          dispatch(Current.setConnectionId(c.id))
```
Comment (jameskerr) [inline src/js/components/ClusterPicker.tsx:49]: Oh nice! Easy

**Review by jameskerr — APPROVED** [review]
Looks awesome. I like the delay, works very well.

**Pull request merged**